rasa_sdk/forms.py

```diff
@@ -412,18 +426,21 @@ def _list_intents(
 
         return self._to_list(intent), self._to_list(not_intent)
 
-    def _log_form_slots(self, tracker: "Tracker") -> None:
+    async def _log_form_slots(self, tracker: "Tracker") -> None:
```

Comment:
couldn't it be that you fetch your slots e.g. from a db?

Reply:
You mean fetching the required slots from the database? I think then I'd still do the retrieval somewhere else to avoid getting them every time the form is triggered.